AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -714,75 +798,20 @@ - (ASLayout *)layoutThatFits:(ASSizeRange)constrainedSize
 - (ASLayout *)layoutThatFits:(ASSizeRange)constrainedSize parentSize:(CGSize)parentSize
 {
   ASDN::MutexLocker l(__instanceLock__);
-
-  if ([self shouldCalculateLayoutWithConstrainedSize:constrainedSize parentSize:parentSize] == NO) {
+  
+  if (_calculatedDisplayNodeLayout->isValidForConstrainedSizeParentSize(constrainedSize, parentSize)) {
```

Comment:
This condition isn't doing the same thing it used to-- compared to the old code in the diff ... but I'm not sure what this branch of the code and the zero size layout it returns is intended to handle.

Could you add a comment describing it if you believe it is an important check?